proposals/trusted_registry_crd.md

```diff
@@ -0,0 +1,128 @@
+# Introduce a custom resource to represent a (trusted) registry
+
+## TOC
+
+- [Abstract](#abstract)
+- [Problem statement](#problem-statement)
+- [Solution](#solutions)
+- [Use cases](#use-cases)
+  - [Starboard Security Operator supervising trusted registries](#starboard-security-operator-supervising-trusted-registries)
+  - [Repository webhooks integrated with Starboard](#repository-webhooks-integrated-with-starboard)
+
+## Abstract
+
+Common requirements for running secure Kubernetes workloads are:
+
+- Pull container images only from trusted registries and/or repositories. For example, one might allow pulling
+  images from private Harbor repository accessible at https://core.harbor.domain/library/nginx:1.16, but deny pull
+  requests to the core.harbor.domain/library/redis:5 or https://docker.io/library/nginx:1.16 repositories.
+- Statically scan container images for critical, high or medium vulnerabilities. In general, these rules are configurable.
+- Use a Kubernetes admission controller to deny workloads with non-compliant container images. For example, one may
+  allow `kubectl run nginx-pod --image=core.harbor.domain/library/nginx:1.16`, but deny
+  `kubectl run redis-pod --image core.harbor.domain/library/redis:5`.
+
+## Problem statement
+
+- Starboard CLI and Starboard Security Operator are able to scan workloads already deployed in a cluster. Hence, an 
+  admission controller cannot use vulnerability reports for compliance checks.
+- The `vulnerabilities.aquasecurity.github.io` resources are scoped to the same namespace as the underlying workload.
+  However, for admission controllers we might want to introduce cluster-scoped resource to represent vulnerability
+  report associated with a given container image digest, for example `registryvulnerabilities.aquasecurity.github.io`.
+- Container images are not first-class citizens in Kubernetes-native world. There’s no built-in resource that represents
+  a container image. Also, there’s no built-in representation of a (trusted) registry.
+- An admission controller has certain performance requirements. Usually we cannot trigger a scanner when the admission
+  controller is invoked for a given Kubernetes API request. Most likely it will take way too long to perform such
+  ad hoc scans.
+
+## Solutions
+
+### Define registries.aquasecurity.github.io resource
+
+First off, introduce a custom resource to represent a (trusted) container registry. For example, we can represent
+a private Harbor registry with the following Kubernetes object:
+
+```yaml
+apiVersion: aquasecurity.github.io/v1alpha1
+kind: Registry
```

Comment:
Is this resource required to create Registry Vulnerability? I think it is unnecessary for some users to specify trusted registries. They might just want to audit vulnerabilities in the image they're trying to deploy.

Reply:
I agree. I think we should be able to create vulnerability reports for the specified image digest and store it at the cluster level without requiring the registry resource.